pipe-cli/src/model/data_storage_wrapper.py

```diff
@@ -407,21 +418,34 @@ def leaf_path(source_path):
         self.path = os.path.abspath(self.path)
 
         if os.path.isfile(self.path):
+            if os.path.islink(self.path) and self.symlinks == AllowedSymlinkValues.SKIP:
+                return []
             return [(FILE, self.path, leaf_path(self.path), os.path.getsize(self.path))]
         else:
             result = list()
+            visited_symlinks = set()
 
-            def list_items(path, parent, root=False):
+            def list_items(path, parent, symlinks, visited_symlinks, root=False):
                 for item in os.listdir(path):
                     absolute_path = os.path.join(path, item)
+                    if os.path.islink(absolute_path) and symlinks != AllowedSymlinkValues.FOLLOW:
+                        if symlinks == AllowedSymlinkValues.SKIP:
+                            continue
+                        if symlinks == AllowedSymlinkValues.FILTER:
+                            symlink_target = os.readlink(absolute_path)
+                            if absolute_path.startswith(symlink_target):
```

Comment:
Could you explain, why do we need this condition?

Reply:
fixed